### PR TITLE
perf-tuning: Add document for column prune

### PR DESCRIPTION
**Pull request by lawyerphx**
### What is changed, added or deleted? (Required)
添加了关于列裁剪的中文文档

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

**Comment by lawyerphx** [comment]
PTAL @winoros

File: column-pruning.md [hunk 1]
```diff
@@ -3,4 +3,67 @@ title: 列裁剪
...
+
+假设表 t 里面有 a b c d 四列，执行如下语句：
+
+```sql
+select a from t where b > 5
+```
+
+在该查询的过程中，t 表实际上只有 a, b 两列会被用到，而 c, d 的数据则显得多余。对应到该语句的查询计划，Selection 算子会用到 b 列，下面接着的 DataSource 算子会用到 a, b 两列，而剩下 c, d 两列则都可以裁剪掉，DataSource 算子在读数据时不需要将它们读进来。
+
+出于上述考量，TiDB 会在逻辑优化阶段进行自上而下的扫描，裁剪不需要的列，减少资源浪费。该扫描过程称作 “列裁剪”，对应逻辑优化规则中的 `columnPruner`。
```
Comment (winoros) [inline column-pruning.md:18]: 可以跟上，如果要关闭这个规则，可以在参照黑名单章节中的关闭方法 之类的

File: column-pruning.md [hunk 2]
```diff
@@ -3,4 +3,67 @@ title: 列裁剪
...
+
+## 算法实现
```
Comment (winoros) [inline column-pruning.md:20]: 算法实现可以直接删去，就不要暴露代码相关的细节了

File: column-pruning.md [hunk 3]
```diff
@@ -3,4 +3,67 @@ title: 列裁剪
...
+
+列裁剪的基本思想在于：对于算子中实际用不上的列，优化器在优化的过程中没有必要保留它们。 对这些列的删除会减少 I/O 资源占用，并为后续的优化带来便利。下面给出一个列重复的例子：
+
+假设表 t 里面有 a b c d 四列，执行如下语句：
+
+```sql
+select a from t where b > 5
+```
+
+在该查询的过程中，t 表实际上只有 a, b 两列会被用到，而 c, d 的数据则显得多余。对应到该语句的查询计划，Selection 算子会用到 b 列，下面接着的 DataSource 算子会用到 a, b 两列，而剩下 c, d 两列则都可以裁剪掉，DataSource 算子在读数据时不需要将它们读进来。
```
Comment (toutdesuite) [inline column-pruning.md:16]:
```suggestion
在该查询的过程中，表 t 实际上只有 a,、b 两列会被用到，不会用到 c、d 列的数据。对应到该语句的查询计划，Selection 算子会用到 b 列，之后 DataSource 算子会用到 a、b 两列。而 c、d 两列则都可以裁剪掉，DataSource 算子在读数据时不需要将它们读进来。
```

File: column-pruning.md (hunk 1 above)
Comment (toutdesuite) [inline column-pruning.md:18]:
```suggestion
出于上述考量，TiDB 会在逻辑优化阶段进行自上而下的扫描，裁剪不需要的列，减少资源浪费。该扫描过程称作“列裁剪”，对应逻辑优化规则中的 `columnPruner`。
```

File: column-pruning.md [hunk 4]
```diff
@@ -3,4 +3,67 @@ title: 列裁剪
...
+
+出于上述考量，TiDB 会在逻辑优化阶段进行自上而下的扫描，裁剪不需要的列，减少资源浪费。该扫描过程称作 “列裁剪”，对应逻辑优化规则中的 `columnPruner`。
+
+## 算法实现
+
+对于逻辑计划中的一个算子而言，它所需要的列分为两种：上层算子计算时所需要的列，本层算子计算时所需要的列。从这个原则出发，优化器可以自顶向下地计算每个算子所需要的列，在遍历的过程中，维护逻辑计划树中的祖先节点给当前算子带来的“列需求”。
```
Comment (toutdesuite) [inline column-pruning.md:22]:
```suggestion
对于逻辑计划中的一个算子，其所需要的列分为两种：上层算子计算时所需要的列，本层算子计算时所需要的列。从这个原则出发，优化器可以自上向下地计算每个算子所需要的列，在遍历的过程中，维护逻辑计划树中的祖先节点给当前算子带来的“列需求”。
```

File: column-pruning.md [hunk 5]
```diff
@@ -3,4 +3,67 @@ title: 列裁剪
...
+
+```goland
+PruneColumns([]*expression.Column) error
+```
+
+函数输入即为施加给当前算子的“列需求”集合。 在普通优化器中，列裁剪会发生在逻辑优化的开始和结尾，[相关代码](https://github.com/pingcap/tidb/blob/902231076d56fee9074e4c7bcd03a0d0f0d88524/planner/core/optimizer.go#L61)。 在 cascades 优化器中，列裁剪会发生在 Preprocessing 阶段，[相关代码](https://github.com/pingcap/tidb/blob/ded862fbebc555de98e230ef57310f9162725a9e/planner/cascades/optimize.go#L118)。
+
+## 常见算子的裁剪方法
+
+本章节将对于用户常见的几个算子进行列裁剪方法简介。默认情况下，优化器不对于当前算子进行列裁剪。
```
Comment (toutdesuite) [inline column-pruning.md:34]:
```suggestion
本章节将介绍几种常见的算子的裁剪方法。默认情况下，优化器不对当前算子进行列裁剪。
```

**Pull request merged**